pkg/session/session.go

```diff
@@ -217,15 +225,10 @@ func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *
 }
 
 func clearCache(logger logr.Logger, sessionKey string) {
-	if cachedSession, ok := sessionCache.Load(sessionKey); ok {
-		s := cachedSession.(*Session)
-
-		logger.Info("performing session log out and clearing session", "key", sessionKey)
-		if err := s.Logout(context.Background()); err != nil {
-			logger.Error(err, "unable to logout session")
-		}
+	if _, ok := sessionCache.Load(sessionKey); ok {
+		logger.Info("clearing the session", "key", sessionKey)
+		sessionCache.Delete(sessionKey)
```

Comment:
Hi @xiujuanx , since sessionCache.Delete() is actually contains `func (m *Map) LoadAndDelete(key any) (value any, loaded bool)` ,  and load&delete in an atomic way, so do we still need the check `sessionCache.Load(sessionKey)` here?

Reply:
Thanks @zhanggbj for your review! 
Indeed, we should not check if the key exists in the cache or not twice. I have changed the code, please help review it again. Thanks!

Reply:
cool, how about https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1949/files#diff-59fbb0bb5dfeeb4f123f1e9e0a3d12c38a60b839e113b29868cb905755639e13R145, this could also be moved out.